scripts/hypothesistooling.py

```diff
@@ -327,3 +327,81 @@ def update_for_pending_release():
         'commit',
         '-m', 'Bump version to %s and update changelog' % (__version__,)
     )
+
+
+def has_no_effect_on_tests(path):
+    """Is this a file which has we can safely assume has no effect on tests?"""
+    # RST files are the input to some tests -- in particular, the
+    # documentation build and doctests.  Both of those jobs are always run,
+    # so we can ignore their effect here.
+    #
+    # IPython notebooks aren't currently used in any tests.
+    if path.endswith(('.rst', '.ipynb')):
+        return True
+
+    # These files exist but have no effect on tests.
+    if path in ('CITATION', 'LICENSE.txt', ):
+        return True
+
+    # All of these files definitely have an effect on tests, and we should
+    # always run tests if any of them have changed.
+    if path.startswith(('src/', 'tests/', 'requirements/', 'setup.py')):
```

Comment:
Can we guarantee that the paths that are passed in are always relative to the repo root? I guess we can because they're always git output, but I feel like this is waiting to bite us on a change.

Reply:
It's unclear to me that this code is useful at all actually given that we default to returning False anyway. It might be better to just delete this part.